frontend/online-shop/src/components/Products.tsx

```diff
@@ -0,0 +1,60 @@
+import { IconButton, ThemeProvider } from "@material-ui/core";
+import { useStyles } from "../styles/tableStyles";
+import InfoIcon from "@material-ui/icons/Info";
+import { Link } from "react-router-dom";
+import axios from "axios";
+import ProductService from "../services/ProductService";
+import React from "react";
+
+export interface ProductEntityProps {
+  id: number;
+  name: string;
+  description: string;
+  category: string;
+  price: number;
+  weight: number;
+  imageUrl: string;
+}
+
+export const TableProducts = (props: ProductEntityProps) => {
+  var productService = new ProductService();
+  var result = productService.getProducts();
+
+  const [data, setData] = React.useState<any[]>([]);
+
+  React.useEffect(() => {
+    result.then((res) => {
```

Comment:
On line 21 you are retrieving a promise and storing it within a variable. This is not usually recommended. You should make the call directly in the `useEffect` hook. Also you should have a `.catch` callback in case the server returns an error and you must handle it. The alternative to `.then` and `.catch` callbacks would be to use `await`.

Reply:
Understood, refactored to use redux instead